### DISCUSSION TITLE
how to access repeater field rows/ fields and main form fields from a function in resource (not closure)

### DISCUSSION THREAD

**Discussion by theo-1** [filamentphp/filament#10945] [Help] [answered]
I have defined an additional function performing some custom tasks in my resource's class. Inside it, I need to be able to refer to each field in each repeater row. 

I see [this](https://github.com/filamentphp/filament/discussions/10399) very useful post but it applies to access within a closure. How do I get a hold of the fields from a function defined directly in the resource class?

**Comment by medfrk** [comment, marked as answer]
Hello, I'm not sure, but I suppose you are calling a function from an element of the form. In this case, you are passing Get $get as a parameter of the function. After that, you can access all the values from your form, including the repeater. For example:
--------------------
// Variable to hold the total price
$total = 0;

// If there are no items in the repeater, return $total as 0
if (!$repeaters = $get('orderItems')) {
    return format_money($total, 'USD', true);
}

// If there are items in the repeater, loop through them and add the sub_totals to $total
foreach ($repeaters as $key => $repeater) {
    $total += $get("orderItems.{$key}.sub_total");
}

$set('total_price', $total);

**Reply by theo-1** [reply]
Will try this but I supppse it would work only in a closure.
